### DISCUSSION TITLE
blocks/cover usage question

### DISCUSSION THREAD

**Discussion by Yeksha** [google/docsy#1720] [Q&A]
I have some questions about using 'blocks/cover.' Can it be used in the documentation/blog
![image](https://github.com/google/docsy/assets/56388318/977fe3c8-d491-433c-95ff-b4494167e2a1)

If it can be used in these places, I found that it didn't generate any `<style>`  code.
``` html
<style>
#{{ $blockID }} {
  background-image: url({{ $promo_image_small.RelPermalink }});
}
@media only screen and (min-width: 1200px) {
  #{{ $blockID }} {
    background-image: url({{ $promo_image_big.RelPermalink }});
  }
}
</style>
```

![image](https://github.com/google/docsy/assets/56388318/9a726f7b-22ec-478d-a0fd-2617d490fb5c)

For example, it generates `<style>` code on the homepage, so it displays
![image](https://github.com/google/docsy/assets/56388318/aef62cef-a376-4855-afa7-8e0b6fb864b2)

**Comment by fekete-robert** [comment]
I think the main intention was to use it on landing pages, so it's possible that using the cover in the docs is not widely tested.

But after a quick check it seems to work for me, if use it within the page is what you want (you might want to wrap the shortcode in a div so you can adjust the margins).

**Comment by Yeksha** [comment]
> I think the main intention was to use it on landing pages, so it's possible that using the cover in the docs is not widely tested.
> 
> But after a quick check it seems to work for me, if use it within the page is what you want (you might want to wrap the shortcode in a div so you can adjust the margins).
Like this? Still not working.
```html
<div class="card-container">
  {{% blocks/cover image_anchor="top" height="auto" %}}
  {{% /blocks/cover %}}
</div>
```

**Comment by fekete-robert** [comment]
Yes, I meant it like that, but just noticed that wrapping it in a div doesn't let you adjust the margins within the shortcode, so you'll have to override them with a more direct css selector. (Something like `.td-content .td-cover`)

But how do you mean it's not working? In the samples you are showing you don't have any content within the cover shortcode, so it only creates an empty gray rectangle (unless you have a properly named image file in the same folder).

**Comment by Yeksha** [comment]
> Yes, I meant it like that, but just noticed that wrapping it in a div doesn't let you adjust the margins within the shortcode, so you'll have to override them with a more direct css selector. (Something like `.td-content .td-cover`)
> 
> But how do you mean it's not working? In the samples you are showing you don't have any content within the cover shortcode, so it only creates an empty gray rectangle (unless you have a properly named image file in the same folder).

Because I like to insert images in the documentation/blog etc.. , I'm confused when using 'blocks/cover' because it doesn't generate style code. As for the image file issue, I wrote an API to fetch random images, and of course, I tested it on the homepage and other places without any problems. Haha, perhaps I need to make some modifications to better fit my usage.

**Comment by chalin** [comment]
Moving this thread to discussions.
